packages/container-query/src/utils.js

```diff
@@ -0,0 +1,31 @@
+export const getBreakpointIndex = (ar = [], v) => {
+  const len = ar.length
+  for (let i = 0; i < len; i++) {
+    if (v <= ar[i]) {
+      return i
+    }
+  }
+  return len
+}
+
+export const em2px = (el, v) =>
+  parseFloat(getComputedStyle(el.parentNode).fontSize || 16) * parseFloat(v)
+
+export const rem2px = v =>
+  parseFloat(getComputedStyle(document.documentElement).fontSize || 16) *
+  parseFloat(v)
+
+const supportedUnits = new Set(['em', 'rem', 'px'])
+
+export const unit2px = (el, val) => {
```

Comment:
I'm not sure I understand the reason for all these utilities and why they would need access to the document/element

Reply:
ResizeObserver returns the width of the element in 'px', but we can define breakpoints using 'em', in order to compare these two values we need to convert em to px. The 'em' size depends on the parent's font-size. Is there any other way to do so?

Reply:
We can use 1em -> 16px, but it might be wrong and the only right way to do so is to use document/element properties.

Reply:
Well, I think I know the reason for the confusion.

useContainerQuery accepts an array of breakpoints based on its own width, not on viewport's width( where we can use css media query).

I thought @junyper meant container query support. There is no CSS support for it and it won't be there any time soon. Let me know is this clear @jxnblk

Reply:
@jxnblk any thoughts on this one? I would like to finish this PR.

Reply:
We can get rid of all these utilities function using CSS Custom Properties:
```
CSS.registerProperty({
  name: '--size',
  syntax: '<length>',
  inherits: false,
  initialValue: 0
});
```
el.style.setProperty('--size', '10em');
el.computedStyleMap().get('--size').value; // returns a correct value in pixels. 

I'll update this PR.